cs/traits.go

```diff
@@ -559,7 +559,7 @@ func urSpec() LRTSpec {
 		// UR gives us 45%/90% of scrapped minerals, versus 33%/80% for races without UR
 		// TODO: Rework scrapping in non-jank way
 		ScrapMineralOffset:           .45 - (1.0 / 3),
-		ScrapMineralOffsetStarbase:   .9 - (1.0 / 3),
```

Comment:
I'm not sure this is correct. The default scrap mineral amount is 1/3rd. Starbases give a "bonus" to that. It needs to be reworked, but the reason it's configured the way it is to treat the UR stuff as a bonus.

Regular race starbases give 80% minerals back, so the default offset for a starbase is 80% - 33%, or a 47% "bonus". UR gets a 57% bonus because you want the total to be 90% (57% + 33% = 90%), so the "bonus" is .9 - 1/3.

Reply:
Yeah, it took me a while to get my head around it and be certain that I got it right. I tested it on my own server and checked the value in the database for ```scrapMineralOffsetStarbase``` and it does end up as 56.66..67 in the race spec for a UR race.

There are two ```scrapMineralOffsetStarbase```'s in the code. One is defined at line 146 of cs/race.go and is a member of ```RaceSpec```. The other is defined at line 78 of cs/traits.go and is a member of ```LRTSpec```.

The way that the code in the PR works for a UR race is:
1. Line 625 of cs/race.go sets ```RaceSpec.scrapMineralOffsetStarbase``` to 47%, which applies to all races.
2. Line 562 of cs/traits.go sets ```LRTSpec.scrapMineralOffsetStarbase``` to 10% for UR races only.
3. Line 748 of cs/race.go increases ```RaceSpec.scrapMineralOffsetStarbase``` by ```LRTSpec.scrapMineralOffsetStarbase```. (This is not changed by the PR). This results in it being 57% for UR races.

The moving parts are quite separated throughout the code. I tried to make minimal changes in what appeared to be the correct places. Perhaps there is a way to rework it that is easier to follow.

I have rebased the code to the latest develop branch.

Reply:
I have added the UR LRT to the ARM test race.

Reply:
I have added the UR LRT to the test case with the race that has ARM.

Reply:
heh, this code is definitely too confusing, but that's on me. :)